packages/freezed/test/integration/json.dart

```diff
@@ -452,7 +452,7 @@ class Generic<T> with _$Generic<T> {
   factory Generic(@DataConverter() T a) = _Generic<T>;
 
   factory Generic.fromJson(Map<String, dynamic> json) =>
-      _$GenericFromJson<T>(json);
+      _$GenericFromJson<T>(json, (DataConverter<T>()).fromJson);
```

Comment:
This currently is a negative change in my opinion.
Two ways to resolve this.
1. Automatically detect JsonConverter annotations that handle generic types matching the surrounding class. (Very difficult)
2. Explicit opt in / out of generic argument json factories, much easier, but unfortunate both if we make it opt in or opt out, and I'm not sure which I would prefer.

Reply:
What is the difficulty with 1 in your opinion?

Reply:
The fact that this is technically a breaking change bothers me. 
I think we'd need to be opt-in.

Reply:
The difficulty is that you have to make sure that **every** parameter in **every** constructor has a type converter for any type recursively containing **any** generic type of the class, even for nested types such as a parameter that has type `List<T>` would need a type converter for `List<T>`.